### ISSUE TITLE
Add YAML attributes to API model classes (like JSON attributes)

### ISSUE THREAD

**Issue by felixfbecker** [tintoy/dotnet-kube-client#22]
I would like to be able to serialize the model classes, e.g. a Deployment, into YAML to save to files. It would be very convenient if the model classes had `YamlMember` attributes for YamlDotNet just like they have `JsonProperty` attributes, so that the properties get serialized into camelCase instead of PascalCase.

**Comment by tintoy** [comment]
Is there a yamldotnet equivalent to CamelCaseJsonContractResolver?

**Comment by tintoy** [comment]
On my phone at the moment, will check when I get to work :)

**Comment by felixfbecker** [comment]
Yeah, it's possible, but I thought it would be nice to have the class know how it should be serialized, given that YAML is the somewhat official serialization format for Kubernetes objects. Ideally I would want to enable an API like this:
```powershell
Get-KubernetesDeployments | ForEach-Object { $_ | ConvertTo-Yaml | Out-File -FilePath ('{0}.{1}.yml' -f $_.Name, $_.Kind) }
```

Where `ConvertTo-Yaml` would "just work", without having to call it as `ConvertTo-Yaml -NamingStrategy [YamlDotNet.Serialization.NamingConvention.CamelCaseNamingConvention]::new()`

**Comment by tintoy** [comment]
Ah, I see what you mean. I was trying to keep the core library's dependency set as small as possible but this seems like a good reason to make an exception :)

I'll have a go at this in around 2 hours time and let you know once there is a build you can test with.

**Comment by felixfbecker** [comment]
It's definitely low priority compared to stuff like auth providers, I can always write my own wrapper function :)

**Comment by tintoy** [comment]
Let's see how this goes :)

https://travis-ci.org/tintoy/dotnet-kube-client/builds/417988968

**Comment by tintoy** [comment]
(once it's done, you can install from the MyGet package feed if you don't mind trying it out)

PS. The authentication provider stuff may take a couple of days - work's looking a little crazy at the start of this week but I'll get to it as soon as I can.

**Comment by felixfbecker** [comment]
If I'm reading the logs correctly, this was published as `2.1.4-unstable0008`? I'm confused because the latest version is `2.1.6`

**Comment by tintoy** [comment]
Yeah - it's a GitVersion issue due to tags. Whatever the build says, that's the source of truth. I'll see what I can do about fixing that version issue but maybe not till this weekend.

**Comment by felixfbecker** [comment]
Do you happen to know how to prevent this error?

```
/Users/felix/src/github.com/felixfbecker/PSKubectl/PSKubectl.csproj : error NU1605: Detected package downgrade: KubeClient from 2.1.6 to 2.1.4-unstable0008. Reference the package directly from the project to select a different version.
/Users/felix/src/github.com/felixfbecker/PSKubectl/PSKubectl.csproj : error NU1605:  PSKubectl -> KubeClient.Extensions.KubeConfig 2.1.6 -> KubeClient (>= 2.1.6)
/Users/felix/src/github.com/felixfbecker/PSKubectl/PSKubectl.csproj : error NU1605:  PSKubectl -> KubeClient (>= 2.1.4-unstable0008)
```

**Comment by tintoy** [comment]
I think it'll resolve itself as long as all the referenced package versions for the various KubeClient.* packages are the same. Could you post the part of your csproj file where the packages are referenced?
